abdiff/core/utils.py

```diff
@@ -124,9 +129,21 @@ def write_to_dataset(
     also be provided, because a schema cannot be extracted or inferred from the lazily
     evaluated generator.  All other types passed do not explicitly require a schema,
     as it can be inferred.
+
+    This function also dynamically sets option 'max_open_files' to align with current OS
+    level configurations open file limits, ensuring not to exceed, which can result in
+    "OSError: [Errno 24] Too many open files".
     """
-    written_files = []
+    # read open file limit and set max_open_files as a fraction
+    if not max_open_files:
+        soft_limit, hard_limit = resource.getrlimit(resource.RLIMIT_NOFILE)
+        max_open_files = max(int(soft_limit * 0.75), 50)
```

Comment:
This ratio of allowing max open files to be about 75% of the OS's limit is somewhat arbitrary, but based on some reading.  100% continued to throw errros, as the margins are too close.  50% worked, but is a decent performance degradation (i.e. half-speed possible of concurrent writes).  75% feels like a sweet spot for the moment of safe buffer to not exceed limit, but still pretty performant.